### PR TITLE
Take file kind from --name when formatting stdin

### PR DESCRIPTION
**Pull request by Julow**
Regression from https://github.com/ocaml-ppx/ocamlformat/pull/1018

This was breaking some editor integration where it's easier to pass the file name than conditionally passing `--intf` or `--impl`, as advertised in the readme (https://github.com/ocaml-ppx/ocamlformat#vim-setup).

With this PR, file kind can be interpreted from the value of `--name`.

On top of https://github.com/ocaml-ppx/ocamlformat/pull/1117

### CODE REVIEW

File: test/cli/stdin_and_name.stdin [hunk 1]
```diff
@@ -0,0 +1 @@
+sample/a.ml
```
Comment (emillon) [inline test/cli/stdin_and_name.stdin]: can you add "Closes #1112" to this commit?

**Comment by emillon** [comment]
I think you can rebase, add the closing bug message, a changelog entry and we can merge and release 0.12.0.

File: test/cli/stdin_and_name.stdin (hunk 1 above)
Comment (Julow) [reply test/cli/stdin_and_name.stdin]: That's not really testing editor integration but I agree that's enough.

File: src/Conf.ml [hunk 2]
```diff
@@ -1721,11 +1727,14 @@ let validate () =
...
-  else if has_stdin && Option.is_none !kind then
+  else if
+    has_stdin && Option.is_none !kind
+    && Option.is_none (Option.bind ~f:kind_of_ext !name)
+  then
     `Error
       ( false
-      , "Must specify at least one of --impl, --intf or --use-file when \
-         reading from stdin" )
+      , "Must specify at least one of --name, --impl or --intf when reading \
```
Comment (Julow) [inline src/Conf.ml]: I removed `--use-file` from the message because it's deprecated now and is equivalent to `--impl`

**Pull request merged**